src/server/createMockServer.ts

```diff
@@ -38,17 +38,24 @@ export default function createMockServer(config: IServerConfig, logger: ILogger)
               metadataOptions: call.metadata.getOptions(),
             };
             let response: any;
-            if (typeof curr.responses === 'function') {
-              response = await curr.responses();
+            if (typeof curr.requests === 'function') {
+              response = await curr.requests();
             } else {
               logger.info(`[${new Date().toISOString()}][${service.name}::${curr.methodName}] Received call:\n`, receivedData)
-              if (config.orderedResponses) {
-                const responseIndex = nextIndex++;
-                nextIndex %= curr.responses.length;
-                response = curr.responses[responseIndex];
+              if (config.randomResponses) {
+                if (config.orderedResponses) {
+                  const responseIndex = nextIndex++;
+                  nextIndex %= curr.requests.length;
+                  response = curr.requests[responseIndex];
+                }else{
+                  const responseIndex = getRandomInt(curr.requests.length);
+                  response = curr.requests[responseIndex];
+                }
               } else {
-                const responseIndex = getRandomInt(curr.responses.length);
-                response = curr.responses[responseIndex];
+                const requestAnswer = curr.requests.find(r => isEqual(r.input, call.request));
+                if(requestAnswer) {
+                  response = requestAnswer.output;
+                }
               }
```

Comment:
`config.randomResponses` is never set in the command, so this is always false (as I mentioned in previous comment)
Can you please change this to:
```typescript
if(config.matchedResponses) {
  // your implementation
} else {
  if (config.orderedResponses) {
    // old ordered logic
  } else {
    // old random logic
  }
}
```